TestVectorsAwsCryptographicMaterialProviders/dafny/TestVectorsAwsCryptographicMaterialProviders/src/VectorsComposition/AllDefaultCmm.dfy

```diff
@@ -0,0 +1,167 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+include "AllAlgorithmSuites.dfy"
+include "AllRawAES.dfy"
+
+module {:options "-functionSyntax:4"} AllDefaultCmm {
+  import opened JSON.Values
+  import opened Wrappers
+  import AllAlgorithmSuites
+  import AllRawAES
+  import SortedSets
+  import StandardLibrary
+
+  import opened UTF8
+  import TestVectors
+  import KeyVectorsTypes = AwsCryptographyMaterialProvidersTestVectorKeysTypes
+  import Types = AwsCryptographyMaterialProvidersTypes
+
+  const StaticNotPlaintextDataKey
+    := KeyVectorsTypes.Static(KeyVectorsTypes.StaticKeyring(
+                                keyId := "no-plaintext-data-key"
+                              ))
+
+  const StaticPlaintextDataKey
+    := KeyVectorsTypes.Static(KeyVectorsTypes.StaticKeyring(
+                                keyId := "static-plaintext-data-key"
+                              ))
+
+  const StaticAlgorithmSuite := AllAlgorithmSuites.AlgorithmSuites.GetESDKSuite(
+                                  AllAlgorithmSuites.Types.ALG_AES_128_GCM_IV12_TAG16_NO_KDF)
+
+  const AesKey := AllRawAES.aesPersistentKeyNames[|AllRawAES.aesPersistentKeyNames| - 1]
+  const RawAesKeyring
+    := KeyVectorsTypes.AES(KeyVectorsTypes.RawAES(
+                             keyId := AesKey,
+                             providerId := "aws-raw-vectors-persistent-" + AesKey
+                           ))
+
+  function SubSets(ec: map<Types.Utf8Bytes, Types.Utf8Bytes>, keys: seq<Types.Utf8Bytes>)
+    : set<map<Types.Utf8Bytes, Types.Utf8Bytes>>
+    requires keys == SortedSets.ComputeSetToOrderedSequence2(ec.Keys, (a, b) => a < b)
+  {
+    if |ec| == 0 then
+      {map[]}
+    else
+      var subsets := SubSets(ec - {keys[0]}, keys[1..]);
+      subsets
+      + (set
+           s <- subsets
+           ::
+             s + map[keys[0] := ec[keys[0]]])
+  }
+
+  const a := UTF8.Encode("a").value
+  const b := UTF8.Encode("b").value
+  const c := UTF8.Encode("c").value
+
+  // Dafny has trouble with complex operations on maps in Java
+  // by decomposing this outside the set comprehension
+  // the translated Java compiles correctly
+  const rootEncryptionContext := map[a := a, b := b]
+  const encryptionContextsToTest := {rootEncryptionContext}
+  const disjointEncryptionContext := map[a := c, b := c, c := c]
+
+  lemma disjointEncryptionContextCorrect()
+    ensures rootEncryptionContext.Values !! disjointEncryptionContext.Values
+    ensures rootEncryptionContext.Items !! disjointEncryptionContext.Items
+  {}
+
+  const SuccessTestingRequiredEncryptionContextKeysReproducedEncryptionContext :=
```

Comment:
i love the set comprehension, but some comments would be nice here

Reply:
I agree. I would like to add this language to the spec and then use `duvet` to line things up